server/api/bot/submit.post.ts

```diff
@@ -5,7 +5,16 @@ const submitBotCodeSchema = z.object({
   code: z.string(),
 });
 
+const rateLimit = new Map<string, number>();
+const RATE_LIMIT_THRESHOLD_IN_MILLIS = 4000;
+
 export default defineEventHandler(async (event) => {
+  const lastRequestTime = rateLimit.get(event.context.user.id);
+
+  if (lastRequestTime && Date.now() - lastRequestTime <= RATE_LIMIT_THRESHOLD_IN_MILLIS) {
+    return { statusCode: 418, statusMessage: "Teapot needs 4 seconds to think about your code" };
```

Comment:
nit: есть специальный http код для таких ситуаций: [429 Too Many Requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429).

И давай константу 4 вынесем в какой-нибудь файл с константами, чтобы у нас был единый источник правды.

```suggestion
    return { statusCode: 429, statusMessage: "Teapot needs 4 seconds to think about your code" };
```